mw-webapp/src/component/table/columns.tsx

```diff
@@ -1,21 +1,52 @@
 import {createColumnHelper, ColumnDef, CellContext} from "@tanstack/react-table";
-import {PlanForTomorrow} from "src/model/report/planForTomorrow/PLanForTomorrow";
-import {WorkDone} from "src/model/report/workDone/WorkDone";
-import {Report} from "src/model/report/Report";
+// import {PlanForTomorrow} from "src/model/report/planForTomorrow/PLanForTomorrow";
+// import {WorkDone} from "src/model/report/workDone/WorkDone";
+// import {Report} from "src/model/report/Report";
 import {Button} from "src/component/button/Button";
-import {ReportService} from "src/service/Report";
+import {ReportService} from "src/service/ReportTest";
 import styles from "src/component/table/columns.module.scss";
+import {DayReport} from "src/model/businessModel/DayReport";
+import {CurrentProblem} from "src/model/businessModel/CurrentProblem";
+import {JobDone} from "src/model/businessModel/JobDone";
+import {PlanForNextPeriod} from "src/model/businessModel/PlanForNextPeriod";
+import {onValue, ref} from "firebase/database";
+import {db} from "src/firebase";
+// import {DayReport as DayReportDTO} from "src/model/firebaseCollection/DayReport";
 
-const columnHelper = createColumnHelper<Report>();
+const columnHelper = createColumnHelper<DayReport>();
 
-const getObjectArrayItem = (arrayItem: WorkDone | PlanForTomorrow, getFullItem?: string) => {
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+const onclick = (index: string, e: any) => {
+  console.log(e.target.parentElement.id);
+  console.log(e.target);
+  if (e.target.parentElement.id === e.target.id) {
+    console.log(true);
+    return ReportService.updateReportToRealTimeDb(index);
+  }
+};
+
+const getObjectArrayItem = (arrayItem: JobDone | PlanForNextPeriod | CurrentProblem, index: number, getFullItem?: string) => {
+  console.log(arrayItem);
+  let qqq;
```

Comment:
Very strange logic. Why can't u use state instead of variables?

Reply:
Yes, I will do but right now this file doesn't render cells, so I will fix it later